src/palace/manager/core/monitor.py

```diff
@@ -810,3 +811,118 @@ class CustomListEntryWorkUpdateMonitor(CustomListEntrySweepMonitor):
 
     def process_item(self, item):
         item.set_work()
+
+
+class TimestampData:
```

Comment:
I moved this here from `metadata_layer` because its only used as part of the monitors, not really the metadata layer, so this seemed like a more appropriate home for it.